.github/workflows/deploy.yml

```diff
@@ -9,8 +9,11 @@ jobs:
         with:
           submodules: true
 
-      - name: Install Rust
-        uses: hecrj/setup-rust-action@v1
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Configure cache
+        uses: Swatinem/rust-cache@v2
```

Comment:
I don't usually look at other project's workflows, so I haven't seen what all the community is using. I see a lot of people using this, but I'm curious if the order of these steps is correct given this known issue at the bottom of the `rust-cache` readme:

> The cache cleaning process currently removes all the files from ~/.cargo/bin that were present before the action ran (for example rustc).

If rustc/cargo/etc are removed and restored from cache, shouldn't the "Setup Rust" step come after to ensure the latest toolchain is always used and to ensure that an older rustc isn't restored from cache?

Reply:
I think the mention of `rustc` in the end is odd, as that's not where the compiler is stored. What they mean is, that any tools installed with `cargo install` will be wiped in the cleanup step.

That is because they include them in the cache, so you don't have to install on the next run again. But, I don't know why they even mention it, is that for the case that you may have any other cleanup step that runs after it and may rely on the binary being present there?

At least that's the only way I can think of that you'd still need the binary :thinking:

Reply:
> If rustc/cargo/etc are removed and restored from cache, [...]

Just to clarify, the toolchain itself is NOT included in the cache. Only extra tools installed with `cargo install`, downloaded crates.io data and git repos, and a few other locations.